vm/systemSmartContracts/delegation.go

```diff
@@ -1621,9 +1650,51 @@ func (d *delegation) withdraw(args *vmcommon.ContractCallInput) vmcommon.ReturnC
 		return vmcommon.UserError
 	}
 
+	err = d.deleteDelegatorIfNeeded(args.CallerAddr, delegator)
+	if err != nil {
+		d.eei.AddReturnMessage(err.Error())
+		return vmcommon.UserError
+	}
+
 	return vmcommon.Ok
 }
 
+func (d *delegation) deleteDelegatorIfNeeded(address []byte, delegator *DelegatorData) error {
+	if d.isOwner(address) {
+		return nil
```

Comment:
this can be easily tested to improve the code coverage

Reply:
It is tested. I added tests for this.

Reply:
already improved test coverage by 12%